doc: add installing section for Connected Home over IP dependencies (#3608)
nrfconnect/sdk-nrf: doc/nrf/gs_installing.rst

      Also see :ref:`zephyr:mac-setup-alts` for additional information.

> If you are interested in building `Connected Home over IP`_ applications, install also the `GN`_ meta-build system to generate files for `Ninja`_.

Reviewer: Maybe reword a bit, since the user probably doesn't care about generating Ninja files, but wants to use the NCS. ;)
```suggestion
If you are interested in building `Connected Home over IP`_ applications, install also the `GN`_ meta-build system to generate the `Ninja`_ files that the |NCS| uses.
```

.. tabs::

   .. group-tab:: Windows

>       The Connected Home over IP project currenlty supports only Linux and macOS.

Reviewer: ```suggestion
      The Connected Home over IP project currently supports only Linux and macOS.
```

>       The Connected Home over IP project currenlty supports only Linux and macOS.

Reviewer: I think we should start by saying that GN is not supported on Windows yet, because that is what the installation is about. And then go on to say that CHIP supports only Linux and macOS.

Author: Ok, I guess it's fixed.

>       The GN, as well as the Connected Home over IP project, do not support Windows yet.

Author: Right, that was simplified too much. I rephrased it a little. Is it ok now?

>       The GN, as well as the Connected Home over IP project, do not support Windows yet.

Reviewer: This I find confusing now.
If GN is actually supported on Windows and the limitation is in CHIP, we should tell users how to install it here and move the information that CHIP is not supported on Windows to the CHIP user guide. Maybe with a note here to prevent users from planning to use CHIP on Windows.

>       The GN, as well as the Connected Home over IP project, do not support Windows yet.

Reviewer 2: But GN is currently only required by CHIP, so if CHIP is not supported on Windows, we shouldn't tell a user to install GN needlessly.

>       The GN, as well as the Connected Home over IP project, do not support Windows yet.

Reviewer: I thought the plan was to just have everybody install it?

>       The GN, as well as the Connected Home over IP project, do not support Windows yet.

Reviewer 2: I'm not aware of such plans. :) I thought that's why there's the following condition at the beginning of the section: "If you are interested in building `Connected Home over IP`_ applications". I believe there are currently no other use cases for GN than building CHIP applications.

>       The GN, as well as the Connected Home over IP project, do not support Windows yet.

Reviewer 3: > I thought the plan was to just have everybody install it?

This was one of the options proposed with @LuDuda and discussed at the tech writers' call, but I understood the way to go is to have a subsection under required tools. So this is the option I went for.

>       The GN, as well as the Connected Home over IP project, do not support Windows yet.

Reviewer 3: However, I wasn't aware GN is supported on Windows... back to drawing board, then.

>       The usage of GN by |NCS|, as well as the Connected Home over IP project, is not supported on Windows yet.
>       The Connected Home over IP project currently supports only Linux and macOS.

Reviewer 3: ```suggestion
      |NCS| does not support the usage of GN on Windows.
      For this reason, the Connected Home over IP project is not yet supported on Windows.
```

   .. group-tab:: Linux

      To install the GN tool, complete the following steps:

      1. Download the GN binary archive and extract it by using the following commands:

         .. parsed-literal::
            :class: highlight

            mkdir ${HOME}/gn && cd ${HOME}/gn
            wget -O gn.zip https://chrome-infra-packages.appspot.com/dl/gn/gn/linux-amd64/+/latest

>             unzip gn.zip gn

Reviewer: What does the "gn" at the end do? Do you mean "-d gn" - but that would give you $HOME/gn/gn ... ?

Author: Yeah, right, I think I've lost `-d` somewhere and that's why I didn't notice the problem. Anyway I removed that option, as in fact `gn.zip` contains only one executable `gn` file, so I think there is not need to create another directory.

            rm gn.zip

      #. Add the location of the GN tool to the system PATH.

>          For example, if you are using ``bashrc``, run the following commands in the directory with extracted executable file:

Reviewer: ```suggestion
         For example, if you are using ``bash``, run the following commands:
```
(It doesn't make a difference where you run the commands.)

         .. parsed-literal::
            :class: highlight

            echo 'export PATH=${HOME}/gn:"$PATH"' >> ${HOME}/.bashrc
            source ${HOME}/.bashrc

   .. group-tab:: macOS

>       To install the GN tool, complete the following steps:

Reviewer: Probably the same changes as for Linux, but I'm not familiar with how it works on macOS.

         pip3 install -U west

> Installing additional toolchains for Connected Home over IP applications development

Reviewer 2: How about "toolchains" -> "tools"?

Author: Ok

====================================================================================

.. note::
   If you are not interested in the `Connected Home over IP`_ applications, you can ommit steps presented in this section.

> `Connected Home over IP`_ project is using `GN`_ meta-build system to generate files for `Ninja`_ and both tools are necessary in order to build an applications.

Reviewer 2: did you mean "in order to build a Connected Home over IP application"?

Author: Yes, isn't it clear enough while whole section applies to the Connected Home over IP?

Reviewer 2: I'm not sure. Probably @greg-fer will have comments on that ;). But please fix the grammar error: "an applications" -> "an application".

However Ninja is also required tool for Zephyr, so it is assumed that it should be installed with the rest of Zephyr's dependencies and GN is the only missing |NCS| requirement.

> .. note::

Reviewer 2: Should we move the note right below the previous one ("If you are not interested in...")?

Author: I don't have strong statement on this, as text is quite short, so second note is not hidden somewhere deep inside the section. I can change it, if you find it useful. Would you prefer to have two notes placed one by one, or both information in the same note?

Reviewer 2: Yes, I think it's better if such notes are at the beginning so that one knows in advance if it's even useful to read the text for him/her.

   Connected Home over IP project supports only Linux and macOS platforms. Windows platform is not available.

Download GN binary archive by entering the following commands:

.. tabs::

   .. group-tab:: Linux

      .. parsed-literal::
         :class: highlight

>          wget -O gn_archive https://chrome-infra-packages.appspot.com/dl/gn/gn/linux-amd64/+/latest

Reviewer 2: The commands will create `gn_archive` and `gn` files in the current directory which is unknown and which a user may not have write permissions to. Should we rather recommend using a known location? Also gn_archive should be deleted once no longer needed. E.g.

```
cd ~
wget -O gn.zip https://chrome-infra-packages.appspot.com/dl/gn/gn/linux-amd64/+/latest
unzip gn.zip gn
rm gn.zip
```
